plugins/createitem.cpp

```diff
@@ -107,126 +100,225 @@ bool makeItem (df::unit *unit, df::item_type type, int16_t subtype, int16_t mat_
+        if (mat_type == -1) {
+            if (tokens[1].empty())
+                out.printerr("You must also specify a growth ID.\n");
+            else
+                out.printerr("The plant you specified has no such growth!\n");
+            out.printerr("Valid growths:%s\n", growths.c_str());
+            return false;
+        }
+    }
+    if (mat_type == -1) {
+        out.printerr("Unrecognized plant ID!\n");
+        return false;
+    }
+    return true;
+}
+
+command_result df_createitem (color_ostream &out, vector<string> &parameters) {
     string item_str, material_str;
-    df::item_type item_type = item_type::NONE;
+    auto item_type = item_type::NONE;
     int16_t item_subtype = -1;
     int16_t mat_type = -1;
     int32_t mat_index = -1;
     int32_t growth_print = -1;
     int count = 1;
     bool move_to_cursor = false;
 
-    if (parameters.size() == 1)
-    {
-        if (parameters[0] == "inspect")
-        {
+    if (parameters.size() == 1) {
+        if (parameters[0] == "inspect") {
             CoreSuspender suspend;
-            df::item *item = Gui::getSelectedItem(out);
+            auto item = Gui::getSelectedItem(out);
             if (!item)
-            {
                 return CR_FAILURE;
-            }
 
             ItemTypeInfo iinfo(item->getType(), item->getSubtype());
             MaterialInfo minfo(item->getMaterial(), item->getMaterialIndex());
             out.print("%s %s\n", iinfo.getToken().c_str(), minfo.getToken().c_str());
             return CR_OK;
         }
-        else if (parameters[0] == "floor")
-        {
+        else if (parameters[0] == "floor") {
             dest_container = -1;
             dest_building = -1;
             out.print("Items created will be placed on the floor.\n");
             return CR_OK;
         }
-        else if (parameters[0] == "item")
-        {
+        else if (parameters[0] == "item") {
             dest_building = -1;
-            df::item *item = Gui::getSelectedItem(out);
-            if (!item)
-            {
+            auto item = Gui::getSelectedItem(out);
+            if (!item) {
                 out.printerr("You must select a container!\n");
                 return CR_FAILURE;
             }
             switch (item->getType())
-            {
-            case item_type::FLASK:
-            case item_type::BARREL:
-            case item_type::BUCKET:
-            case item_type::ANIMALTRAP:
-            case item_type::BOX:
-            case item_type::BAG:
-            case item_type::BIN:
-            case item_type::BACKPACK:
-            case item_type::QUIVER:
-                break;
-            case item_type::TOOL:
-                if (item->hasToolUse(tool_uses::LIQUID_CONTAINER))
-                    break;
-                if (item->hasToolUse(tool_uses::FOOD_STORAGE))
-                    break;
-                if (item->hasToolUse(tool_uses::SMALL_OBJECT_STORAGE))
+            {   using namespace df::enums::item_type;
```

Comment:
I've never seen this syntax before -- a `using` *inside* of a switch statement. useful.